### ISSUE TITLE
Unable to upgrade Debian packages

### ISSUE THREAD

**Issue by andrewsomething** [owncloud/core#16949]
Hi!

This seems to be related to (if not the same as) #16330 and #16007 even though both are marked as closed. `/var/www/owncloud/.htaccess` was moved to the owncloud-server package from the owncloud-config-apache package. A Breaks/Replace was added on `owncloud-config-apache (<< 8.0.3)` but this doesn't seem to be enough. This is still causing an error when attempting to upgrade, in my case from 8.0.2-5 to 8.0.4-0.

The error:

```
dpkg: error processing archive /var/cache/apt/archives/owncloud-server_8.0.4-0_all.deb (--unpack):
 trying to overwrite '/var/www/owncloud/.htaccess', which is also in package owncloud-config-apache 8.0.4-0
```

So it looks like owncloud-config-apache didn't just install the `.htaccess` file, but also declared it a conffile. These are special files, and they do not get removed when a package is uninstalled unless `--purge` is used. So while that file is not in the package anymore, for anyone upgrading from a previous version it is still "owned" by owncloud-config-apache.

See: https://www.debian.org/doc/debian-policy/ap-pkg-conffiles.html

On a system with owncloud-config-apache 8.0.2-5 installed, removing it confirms this:

```
# apt-get remove owncloud-config-apache
[snip...]
Removing owncloud-config-apache (8.0.2-5) ...
root@another-owncloud:~# dpkg-query --showformat='${Conffiles}\n' --show owncloud-config-apache
 /etc/apache2/conf-available/owncloud.conf b8c04bb565d60025727d9b89af6e03f5
 /var/www/owncloud/.htaccess b84088cc7a360c7f1aba3141b4f17e64
 /etc/apache2/conf-available/owncloud.conf newconffile
# file /var/www/owncloud/.htaccess
/var/www/owncloud/.htaccess: ASCII text
# dpkg -S /var/www/owncloud/.htaccess
owncloud-config-apache: /var/www/owncloud/.htaccess
```

This blog post describes how obsolete conffile's should be removed in the package's maintainer script:

https://raphaelhertzog.com/2010/10/07/the-right-way-to-remove-an-obsolete-conffile-in-a-debian-package/

owncloud-server and owncloud-config-apache versions:

```
owncloud-server:
  Installed: 8.0.2-5
  Candidate: 8.0.4-0
  Version table:
     8.0.4-0 0
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_14.04/  Packages
 *** 8.0.2-5 0
        100 /var/lib/dpkg/status

owncloud-config-apache:
  Installed: 8.0.2-5
  Candidate: 8.0.4-0
  Version table:
 *** 8.0.4-0 0
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/community/xUbuntu_14.04/  Packages
        100 /var/lib/dpkg/status
```

I wasn't able to find the packaging in version control, so hopefully this is the right place to report these things.

**Comment by DeepDiver1975** [comment]
@jnweiger THX

**Comment by jnweiger** [comment]
dpkg-maintscript-helper rm_conffile is a great hint. Thanks!

**Comment by jnweiger** [comment]
dpkg-maintscript-helper rm_conffile added in 
https://build.opensuse.org/package/show/isv:ownCloud:community:8.0:testing/owncloud
version 8.0.4-1

**Comment by andrewsomething** [comment]
@jnweiger I tried the packages in the testing repository and can confirm that it now upgrades successfully.

```
root@oldcloud:~# apt-cache policy owncloud-config-apache 
owncloud-config-apache:
  Installed: 8.0.4-1
  Candidate: 8.0.4-1
  Version table:
 *** 8.0.4-1 0
        500 http://download.opensuse.org/repositories/isv:/ownCloud:/community:/8.0:/testing/xUbuntu_14.04/  Packages
        100 /var/lib/dpkg/status
root@oldcloud:~# dpkg-query --showformat='${Conffiles}\n' --show owncloud-config-apache
 /etc/apache2/conf-available/owncloud.conf 96ebac4c9fc874d8c8dfe362f16082f1
 /etc/apache2/conf-available/owncloud.conf b8c04bb565d60025727d9b89af6e03f5
root@oldcloud:~# cat /var/log/apt/term.log | grep htaccess
Moving obsolete conffile /var/www/owncloud/.htaccess out of the way...
Removing obsolete conffile /var/www/owncloud/.htaccess ...
```

Thanks for the prompt fix!

**Comment by DeepDiver1975** [comment]
@jnweiger @andrewsbrown case closed

**Comment by jnweiger** [comment]
fix will be part of 8.0.5 packages
